src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentBindLoweringPass.cs

```diff
@@ -521,6 +485,151 @@ private IntermediateNode[] RewriteUsage(IntermediateNode parent, TagHelperProper
+                Content = $"{ComponentsApi.BindMethods.GetValue}(",
+                Kind = TokenKind.CSharp
+            });
+            valueExpressionTokens.Add(original);
+            if (!string.IsNullOrEmpty(format?.Content))
+            {
+                valueExpressionTokens.Add(new IntermediateToken()
+                {
+                    Content = ", ",
+                    Kind = TokenKind.CSharp,
+                });
+                valueExpressionTokens.Add(format);
+            }
+            valueExpressionTokens.Add(new IntermediateToken()
+            {
+                Content = ")",
+                Kind = TokenKind.CSharp,
+            });
+
+            // Now rewrite the content of the change-handler node. There are two cases we care about
+            // here. If it's a component attribute, then don't use the 'CreateBinder' wrapper. We expect
+            // component attributes to always 'match' on type.
+            //
+            // The really tricky part of this is that we CANNOT write the type name of of the EventCallback we
+            // intend to create. Doing so would really complicate the story for how we deal with generic types,
+            // since the generic type lowering pass runs after this. To keep this simple we're relying on
+            // the compiler to resolve overloads for us.
+            //
+            // EventCallbackFactory.CreateInferred(this, __value => <code> = __value, <code>)
+            //
+            // For general DOM attributes, we need to be able to create a delegate that accepts UIEventArgs
+            // so we use 'CreateBinder'
+            //
+            // EventCallbackFactory.CreateBinder(this, __value => <code> = __value, <code>) OR
+            // EventCallbackFactory.CreateBinder(this, __value => <code> = __value, <code>, <format>)
+            //
+            // Note that the linemappings here are applied to the value attribute, not the change attribute.
+
+            string changeExpressionContent;
+            if (changeAttribute == null && format == null)
+            {
+                // DOM
+                changeExpressionContent = $"{ComponentsApi.EventCallback.FactoryAccessor}.{ComponentsApi.EventCallbackFactory.CreateBinderMethod}(this, __value => {original.Content} = __value, {original.Content})";
+            }
+            else if (changeAttribute == null && format != null)
+            {
+                // DOM + format
+                changeExpressionContent = $"{ComponentsApi.EventCallback.FactoryAccessor}.{ComponentsApi.EventCallbackFactory.CreateBinderMethod}(this, __value => {original.Content} = __value, {original.Content}, {format.Content})";
+            }
+            else
+            {
+                // Component
+                changeExpressionContent = $"{ComponentsApi.EventCallback.FactoryAccessor}.{ComponentsApi.EventCallbackFactory.CreateInferredMethod}(this, __value => {original.Content} = __value, {original.Content})";
```

Comment:
Note: this is another really thorny layering problem. The fact that `bind-...` generates code during this pass instead of creating custom node types creates a bunch of other hard problems to solve. 

I did another overloading-based workaround here to avoid the problem where I can't write `T`. The most correct fix would take a collossal amount of revisiting existing work.

Reply:
I see. Is the goal to eventually bake the component specific IR more into the default set of IR nodes or is it to use extension nodes. I'd assume the prior but I'm curious.

Reply:
yes, the goal is just build everything components as first-class.

Reply:
I actually tried working through this for 5-6 hours, but I needed to define an inheritance hierachy of parameter attribute nodes, and I needed to basically start over with the code generation for attributes, or duplicate a massive amount of code. I think fixing this properly is a 3 days of work for someone. 

I basically proved to myself that we know the right design for something like this, but if we don't follow it then that doesn't help 😆